tests/openssl/CMakeLists.txt

```diff
@@ -0,0 +1,512 @@
+  copy_test_data(${OPENSSL_DIR}/test/certs/embeddedSCTs3.pem
+                 ${CMAKE_CURRENT_BINARY_DIR}/certs/embeddedSCTs3.pem)
+  copy_test_data(${OPENSSL_DIR}/test/certs/embeddedSCTs3.sct
+                 ${CMAKE_CURRENT_BINARY_DIR}/certs/embeddedSCTs3.sct)
+  copy_test_data(${OPENSSL_DIR}/test/certs/embeddedSCTs3_issuer.pem
+                 ${CMAKE_CURRENT_BINARY_DIR}/certs/embeddedSCTs3_issuer.pem)
+  copy_test_data(${OPENSSL_DIR}/test/certs/leaf.pem
+                 ${CMAKE_CURRENT_BINARY_DIR}/certs/leaf.pem)
+  copy_test_data(${OPENSSL_DIR}/test/certs/subinterCA.pem
+                 ${CMAKE_CURRENT_BINARY_DIR}/certs/subinterCA.pem)
+  copy_test_data(${OPENSSL_DIR}/test/ct/tls1.sct
+                 ${CMAKE_CURRENT_BINARY_DIR}/ct/tls1.sct)
+
+  # Add a custom target that ensures the test files are copied at build time.
+  # The presence of the target indicates that we can enable the depending tests
+  # (i.e., evp_test and ct_test). If the target is not present, these tests are
+  # skipped.
+  add_custom_target(openssl_test_data ALL DEPENDS ${test_data_list})
+endif ()
+
+# The paths to the arguments used by the tests.
+set(OPENSSL_APPS_DIR ${OPENSSL_DIR}/apps)
+set(OPENSSL_TEST_DIR ${OPENSSL_DIR}/test)
+set(TEST_DATA_DIR ${CMAKE_CURRENT_BINARY_DIR})
+if (WIN32)
+  # Convert the "C:/" to "C/" so that the path can be used by hostfs on Windows.
+  string(REPLACE ":" "" OPENSSL_APPS_DIR "${OPENSSL_APPS_DIR}")
+  string(REPLACE ":" "" OPENSSL_TEST_DIR "${OPENSSL_TEST_DIR}")
+  string(REPLACE ":" "" TEST_DATA_DIR "${TEST_DATA_DIR}")
+endif ()
+
+# Add buildtests.
+file(STRINGS "tests.supported.buildtest" buildtests)
+foreach (testcase ${buildtests})
+  get_testcase_name(${testcase} name "../../3rdparty/openssl/openssl/test/")
+  add_enclave_test(tests/openssl/${name} openssl_host openssl-${name}_enc
+                   ${name})
+endforeach (testcase)
+
+# Helper for adding tests with arguments.
+macro (add_openssl_test TEST_NAME NAME)
+  add_enclave_test(tests/openssl/${TEST_NAME} openssl_host openssl-${NAME}_enc
+                   ${TEST_NAME} ${ARGN})
+endmacro (add_openssl_test)
+
+# Add the rest of the tests.
+file(STRINGS "tests.supported" openssltests)
+foreach (testcase ${openssltests})
+  get_testcase_name(${testcase} name "../../3rdparty/openssl/openssl/test/")
+
+  if ("${name}" STREQUAL asynciotest)
+    add_openssl_test("${name}" "${name}" ${OPENSSL_APPS_DIR}/server.pem
+                     ${OPENSSL_APPS_DIR}/server.pem)
+    continue()
+  endif ()
+
+  if ("${name}" STREQUAL clienthellotest)
+    add_openssl_test("${name}" "${name}" ${OPENSSL_TEST_DIR}/session.pem)
+    continue()
+  endif ()
+
+  if ("${name}" STREQUAL cmsapitest)
+    add_openssl_test(
+      "${name}" "${name}" ${OPENSSL_TEST_DIR}/certs/servercert.pem
+      ${OPENSSL_TEST_DIR}/certs/serverkey.pem)
+    continue()
+  endif ()
+
+  # The following tests fail with DEBUG_MALLOC because of the use of setenv.
```

Comment:
What is the issue betwen DEBUG_MALLOC and setenv? 
CC: @nonpolarity

Reply:
`setenv` would allocate memory but is not freed entirely. I looked into that and it seems has to do with the musl implementation.

Reply:
Thanks for the investigation. We might likely have to put workarounds in the test (in a future PR). We are trying to get to a point where all tests under DEBUG_MALLOC.